jf_agent/agent_logging.py

```diff
@@ -217,6 +219,50 @@ def filter(self, record: LogRecord) -> bool:
         return not record.__dict__.get(AGENT_LOG_TAG, False)
 
 
+def configure_structlog() -> None:
+    """
+    Configures rendering of log messages within structlog, which then passes a fully formatted string to
+    the logging module for output.
+    """
+    structlog.configure(
+        processors=[
+            structlog.contextvars.merge_contextvars,
+            structlog.stdlib.filter_by_level,
+            structlog.stdlib.add_logger_name,
+            structlog.stdlib.add_log_level,
+            structlog.stdlib.PositionalArgumentsFormatter(),
+            structlog.processors.TimeStamper(fmt="iso"),
+            structlog.processors.StackInfoRenderer(),
+            structlog.processors.format_exc_info,
+            structlog.processors.UnicodeDecoder(),
+            structlog.processors.CallsiteParameterAdder(
+                {
+                    structlog.processors.CallsiteParameter.FILENAME,
+                    structlog.processors.CallsiteParameter.FUNC_NAME,
+                    structlog.processors.CallsiteParameter.LINENO,
+                }
+            ),
+            structlog.processors.JSONRenderer(),
+        ],
+        wrapper_class=structlog.stdlib.BoundLogger,
+        logger_factory=structlog.stdlib.LoggerFactory(),
+        cache_logger_on_first_use=True,
+    )
+
+
+def bind_context_vars(
+    run_mode: str, company_slug: Optional[str], upload_time: str,
```

Comment:
Type hinting `company_slug` as Optional since the value passed to that param comes from `.get()`